src/cloudimagedirectory/filter/filter.py

```diff
@@ -15,6 +19,14 @@ def get_utc_datetime(date_string: str) -> pd.Timestamp:  # type: ignore[no-any-u
 def FilterImageByFilename(word: str) -> Callable:
     """Filter images by filename."""
     print("filter images by filename: " + word)
+
+    meter.create_counter(
+        name="filtered_image_by_filename_counter",
+        description="Counts the number of filtered images by filename",
+        unit="1",
+    )
+    # TODO: Call method every time the lambda skips an element.
+    # filtered_image_by_filename_counter.add(1, {"word", word})
     return lambda data: [d for d in data if not d.filename.lower().__contains__(word.lower())]
```

Comment:
I am afraid you will have to replace the lambda functions and get rid of the list generators. I don't think there is a nice trick that would allow you to do what you need. You can create an inner function instead and return that one.

```
def f(data):
    result = []
    for d in data:
        ...
    return result

return f
```

Reply:
@poncovka , I tried to replace the lambda functions with inner functions. 
Could you have a look again, if it makes sense how I did it and if it is in a "python way" :sweat_smile: ?